teuthology/task/kernel.py

```diff
@@ -906,12 +907,25 @@ def grub2_kernel_select_generic(remote, newversion, ostype):
     remote.run(args=['sudo', mkconfig, '-o', grubconfig, ])
     grub2conf = teuthology.get_file(remote, grubconfig, True)
     entry_num = 0
-    for line in grub2conf.split('\n'):
-        if line.startswith('menuentry'):
-            if newversion in line:
+    if '\nmenuitem ' not in grub2conf:
+        # okay, do the newer (el8) grub2 thing
+        proc = remote.run(args=['sudo', '/bin/ls', '/boot/loader/entries'],
+                          stdout=StringIO())
+        grub2conf = proc.stdout.getvalue()
```

Comment:
```
grub2conf = remote.run('sudo /bin/ls /boot/loader/entries')
```

Reply:
fixed.  the change below will conflict tho right?

Reply:
I think you meant remote.sh here.

Reply:
yeah :-) sorry, I meant `remote.sh` :-)